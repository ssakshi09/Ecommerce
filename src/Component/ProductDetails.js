import React from "react";
import { Typography } from "@material-ui/core";

const ProductDetails = ({ product }) => {
  return (
    <div>
      <Typography gutterBottom variant="h5" component="h2"></Typography>
      <Typography variant="body2" color="textSecondary" component="p">
        Rating: ${product.rating}
      </Typography>
      <Typography variant="body1" component="p">
        {product.description}
      </Typography>
    </div>
  );
};

export default ProductDetails;
