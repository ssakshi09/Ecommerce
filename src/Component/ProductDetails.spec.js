import { render } from "@testing-library/react";
import ProductDetails from "./ProductDetails";


describe("Product Details", () => {
  const product = {
    id: 1,
    description: "An apple mobile which is not sing like apple",
    rating: 4.69,
  };
  it("Should display Rating and Description", () => {
    const wrapper = render(<ProductDetails product={product} />);
    expect(wrapper).toMatchSnapshot();
  });
});
