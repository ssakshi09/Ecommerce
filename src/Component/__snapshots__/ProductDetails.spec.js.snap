// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Details Should display Rating and Description 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        />
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          Rating: $
          4.69
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          An apple mobile which is not sing like apple
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      />
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Rating: $
        4.69
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        An apple mobile which is not sing like apple
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
